lordpengwin/muzak
# Add support for claudia.js deployment to AWS Lambda

Hey @lordpengwin and @MikeDeSantis! I think this will greatly improve the development workflow for muzak. 

I stumbled on [claudia.js](https://claudiajs.com/) the other day and it's super awesome. It basically just lets you run a command to push your latest changes to AWS Lambda instead of having to manually create a zip file and upload it yourself. Saves a bunch of time.  This also adds a few other things that I'll call out below. 

### What this does
- Adds support for claudia deployments
  - Adds claudia as a local npm dependency
  - Updates the README with usage instructions
- Adds an `.nvmrc` file to encourage using the same version of node as AWS Lambda
- Removes the check for `applicationId` (I'll explain why I did this in a comment below)
- Adds a basic test fixture that you can run via `npm run test`. In the future, we can add tests for everything. 

### How To Test Claudia 
- Follow the instructions in the README. Pay particularly attention to setting up your AWS creds. 
- You guys already have Lambda functions set up for muzak, so to make it easier be sure to pass the name of your function to the `claudia create` command. 
- Run `npm test` and you should get a proper response. 
- Make a change in `muzak.js` and run `claudia update` and like magic your changes are pushed to AWS. ✨ 

What do you think?

PS - The commit history is kind of garbage since my old commits for adding `package.json` weren't merged into the repo before. Apologies for not creating the PR for that against the original repo!
